src/DiscGolfScorecard.java

```diff
@@ -0,0 +1,378 @@
+     * @param startingHole
+     *            the first hole returned in scorecard
+     * @param pars
+     *            array of the pars for each hole
+     * @requires scorecardSize > 0 and 0 < startingHole <= scorecardSize and
+     *           scorecardSize = |pars| and for all x: int where (x is in pars)
+     *           x >= 0
+     * @ensures |$this.scorecard| = scorecardSize and this.scorecard[is a
+     *          scorecard with holes with the specified pars]
+     */
+    private void createNewRep(int scorecardSize, int startingHole, int[] pars) {
+        this.scorecard = new DiscGolfHole[scorecardSize];
+        this.currentHole = startingHole;
+
+        //Add a hole to each element of the array.
+        for (int i = 0; i < scorecardSize; i++) {
+            this.scorecard[i] = new DiscGolfHole(pars[i]);
+        }
+    }
+
+    /**
+     * Creator of inital representation.
+     *
+     * @param scorecardSize
+     *            number of holes in scorecard
+     * @param startingHole
+     *            the first hole returned in scorecard
+     * @param pars
+     *            array of the pars for each hole
+     * @param distances
+     *            array of the distances for each hole
+     * @requires scorecardSize > 0 and 0 < startingHole <= scorecardSize and
+     *           scorecardSize = |pars| = |distances| and for all x: int where
+     *           (x is in pars or distances) x >= 0
+     * @ensures |$this.scorecard| = scorecardSize and this.scorecard[is a
+     *          scorecard with holes with the specified pars and distances]
+     */
+    private void createNewRep(int scorecardSize, int startingHole, int[] pars,
+            int[] distances) {
+        this.scorecard = new DiscGolfHole[scorecardSize];
+        this.currentHole = startingHole;
+
+        //Add a hole to each element of the array.
+        for (int i = 0; i < scorecardSize; i++) {
+            this.scorecard[i] = new DiscGolfHole(pars[i], distances[i]);
+        }
+    }
+
+    /**
+     * Creator of inital representation.
+     *
+     * @param scorecardSize
+     *            number of holes in scorecard
+     * @param startingHole
+     *            the first hole returned in scorecard
+     * @param distances
+     *            array of the distances for each hole
+     * @param pars
+     *            array of the pars for each hole
+     * @param strokes
+     *            array of the strokes for each hole
+     * @requires scorecardSize > 0 and 0 < startingHole <= scorecardSize and
+     *           scorecardSize = |pars| = |distances| = |strokes| and for all x:
+     *           int where (x is in pars or distances or strokes) x >= 0
+     * @ensures |$this.scorecard| = scorecardSize and this.scorecard[is a
+     *          scorecard with holes with the specified pars, distances, and
+     *          strokes]
+     */
+    private void createNewRep(int scorecardSize, int startingHole, int[] pars,
```

Comment:
I think having this many createNewReps is fine. It's an implementation detail. However, you may find simplifying the design to be a little easier to maintain.